core/services/network_checker.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require_relative 'log_storage'
+require_relative 'machine_configurator'
+
+# This class checks the health of network resources
+module NetworkChecker
+  RESOURCES = %w[https://github.com
+                 https://www-eu.apache.org/
+                 https://nodejs.org/
+                 http://prdownloads.sourceforge.net/].freeze
+
+  # Check all resources for availability
+  # @return [Boolean] true if all resources available
+  def self.resources_available?(machine_configurator, machine, logger)
+    tool = if check_available_tool?('curl', machine_configurator, machine, logger)
+             'curl'
+           else
+             'wget'
+           end
+    random_resources.each do |resource|
```

Comment:
There is a special iterator [`all?`](https://ruby-doc.org/core-2.7.0/Array.html#method-i-all-3F). In the block you must return true or false. If any check fails, then the result will be false. If all return true, then all is fine.

Switch from each to all? in this situation.